pkg/networkpolicy/adminnetworkpolicy.go

```diff
@@ -0,0 +1,282 @@
+		// ANPs with no ingress rules do not affect ingress traffic.
+		for _, rule := range policy.Spec.Ingress {
+			// Ports allows for matching traffic based on port and protocols.
+			// This field is a list of destination ports for the outgoing egress traffic.
+			// If Ports is not set then the rule does not filter traffic via port.
+			if rule.Ports != nil {
+				if !evaluateAdminNetworkPolicyPort(*rule.Ports, pod, port, protocol) {
+					continue
+				}
+			}
+			// To is the List of destinations whose traffic this rule applies to.
+			// If any AdminNetworkPolicyEgressPeer matches the destination of outgoing
+			// traffic then the specified action is applied.
+			// This field must be defined and contain at least one item.
+			for _, from := range rule.From {
+				// Exactly one of the selector pointers must be set for a given peer. If a
+				// consumer observes none of its fields are set, they must assume an unknown
+				// option has been specified and fail closed.
+				if from.Namespaces != nil {
+					if c.namespaceSelector(from.Namespaces, pod) {
+						return rule.Action
+					}
+				}
+
+				if from.Pods != nil {
+					if c.namespaceSelector(&from.Pods.NamespaceSelector, pod) &&
+						podSelector(&from.Pods.PodSelector, pod) {
+						return rule.Action
+					}
+				}
+			}
+
+		}
+	}
+
+	return npav1alpha1.AdminNetworkPolicyRuleActionPass
+}
+
+// namespaceSelector return true if the namespace selector matches the pod
+func (c *Controller) namespaceSelector(selector *metav1.LabelSelector, pod *v1.Pod) bool {
+	nsSelector, err := metav1.LabelSelectorAsSelector(selector)
+	if err != nil {
+		return false
+	}
+
+	namespaces, err := c.namespaceLister.List(nsSelector)
+	if err != nil {
+		return false
+	}
+
+	for _, ns := range namespaces {
+		if pod.Namespace == ns.Name {
+			return true
+		}
+	}
+	return false
+}
+
+// podSelector return true if the pod selector matches the pod
+func podSelector(selector *metav1.LabelSelector, pod *v1.Pod) bool {
+	podSelector, err := metav1.LabelSelectorAsSelector(selector)
+	if err != nil {
+		return false
+	}
+	return podSelector.Matches(labels.Set(pod.Labels))
+}
+
+// nodeSelector return true if the node selector matches the pod
+func (c *Controller) nodeSelector(selector *metav1.LabelSelector, pod *v1.Pod) bool {
+	nodeSelector, err := metav1.LabelSelectorAsSelector(selector)
+	if err != nil {
+		return false
+	}
+	nodes, err := c.namespaceLister.List(nodeSelector)
+	if err != nil {
+		return false
+	}
+
+	for _, node := range nodes {
+		if pod.Spec.NodeName == node.Name {
+			return true
+		}
+	}
+	return false
+}
+
+// getAdminNetworkPoliciesForPod returns the list of Admin Network Policies matching the Pod
+// The list is ordered by priority, from higher to lower.
+func (c *Controller) getAdminNetworkPoliciesForPod(pod *v1.Pod) []*npav1alpha1.AdminNetworkPolicy {
```

Comment:
I find it slightly confusing, that `getNetworkPoliciesForPod` only returns policies from the same namespace, and then it checks if they really apply to the pod in `evaluator`, but here we check it all at once. 
Maybe we could change `getNetworkPoliciesForPod` to also check podSelector?

Reply:
this is evaluating the Spec.Subject specifically , can you expand what else need to be evaluated?

Reply:
I am comparing similar methods for networkpolicy and adminnetworkpolicy.
This one for adminnetworkolicy is actually good, but networkpolicy `getNetworkPoliciesForPod` method doesn't filter based on podSelector. So can we add that filtering here https://github.com/kubernetes-sigs/kube-network-policies/blob/main/pkg/networkpolicy/networkpolicy.go#L19-L23 and remove from here https://github.com/kubernetes-sigs/kube-network-policies/blob/main/pkg/networkpolicy/networkpolicy.go#L78-L87?

Reply:
great review

Reply:
@npinaeva do you mind sending a PR for that?